### DISCUSSION TITLE
Prioritize Most Recently Used windows in Alt-Tab interface

### DISCUSSION THREAD

**Discussion by BillyCroan** [linuxmint/discussions#317] [Ideas]
It is a human behavioral truth, that when we have many tools, there is likely to be a small few that get used more than others.  And the more likely than not, the next tool you will use will be the last one you used.  Windows on a graphical desktop are no exception.

Many desktop environments (including essentially all other distributions, operating systems, and window managers) reduce the number of clicks and keystrokes required of the user by taking advantage of this truth.  Mint does not.

I propose improving Cinnamon, with a new setting in the "Alt-Tab" tab of the "Windows" control panel interface.
![image](https://github.com/linuxmint/discussions/assets/37587/6144f4c1-4848-4411-b9ad-ba22562a3f3c)

Most Recently Used, would keep a datetime of the last time the user interacted with each window.  And when using Alt Tab, that list is consulted.  The currently active window, which would always be on the top of that list, is displayed at the end of the list.  And all others are displayed in order, with the default being the second most recently used.

Classic, would remain the default, and continue to do alt-tab however it is Cinnamon does currently.  Alphabetical?  Date of install?  time opened?  lunar cycle of forth commit to SVN tree alphabetically.

**Comment by anaximeno** [comment]
It already works this way or like this way I guess, the only different thing than what you suggests is that the current active window is the first on the alt-tab switcher, but, generally, the ordering is the way you suggest I believe.

**Comment by BillyCroan** [comment]
Current one being first is fine if then the second one is selected by default.

If so, would you say that: (without looking at the screen) I should be able to take a screen shot.  then press Alt tab, then release the keyboard.  Then press Alt tab again and release the keyboard and take a second screen shot, and they should look the same? (Absent sometime new and unrelated opening or closing during that time)

**Reply by anaximeno** [reply]
That doesn't seem like a good example of what you are suggesting I guess (unless I didn't understand well), we are not taking about screenshots looking the same, are we? Shouldn't the example be only about the order of the applications on the alt-tab switcher? The screenshot example could not work because the default screenshot app on Mint after taking the screenshot, it closes the app and open the screenshot image, so as far as I remember you wouldn't take two screenshots with the same instance.

**Reply by BillyCroan** [reply]
You're right.  screenshots are a bad example.  I made a video, documenting the misbehavior I am observing, and I think it is going to work a lot better than describing it in writing. https://youtu.be/PQDY10vYZAc

If you think of a most recently used "stack" where every time you use a window it goes to the top of the stack, it appears that firefox, and evince too, for some reason, get stuck at a lower place in the stack.  Not all windows experience this problem, but I use firefox and evince a lot.

**Reply by anaximeno** [reply]
The video is private so I'm not able to see it. You can try to drag and drop the video from your computer here in the comment section if it is not too big.

**Comment by averter** [comment]
+1 on this. I absolutely agree that 1) the currently used window should be the leftmost, and 2) the remaining windows after it should be ordered in terms of how recently they have been used/accessed. 
Strangely, linux mint used to work like this but it seems to have been changed recently.
